examples/safe-math-lib/1.0.0.json

```diff
@@ -3,80 +3,53 @@
   "version": "1.0.0",
   "package_name": "safe-math-lib",
   "sources": {
-    "./contracts/SafeMathLib.sol": "ipfs://QmVN1p6MmMLYcSq1VTmaSDLC3xWuAUwEFBFtinfzpmtzQG"
+    "./contracts/SafeMathLib.sol": "ipfs://QmNQMViTJ6dne4GNnTwHKnDwXxRaGAc5i2gNuU7T2G21uA"
   },
   "contract_types": {
     "SafeMathLib": {
       "deployment_bytecode": {
-        "bytecode": "0x606060405234610000575b60a9806100176000396000f36504062dabbdf050606060405260e060020a6000350463a293d1e88114602e578063e6cb901314604c575b6000565b603a600435602435606a565b60408051918252519081900360200190f35b603a6004356024356088565b60408051918252519081900360200190f35b6000828211602a57508082036081566081565b6000565b5b92915050565b6000828284011115602a57508181016081566081565b6000565b5b9291505056"
+        "bytecode": "0x604c602c600b82828239805160001a60731460008114601c57601e565bfe5b5030600052607381538281f30073000000000000000000000000000000000000000030146080604052600080fd00a165627a7a7230582083e86e46995b268179a35d3d89e28a4913a8104c518b63bc2b1aee04f9c0b6610029"
       },
       "runtime_bytecode": {
-        "bytecode": "0x6504062dabbdf050606060405260e060020a6000350463a293d1e88114602e578063e6cb901314604c575b6000565b603a600435602435606a565b60408051918252519081900360200190f35b603a6004356024356088565b60408051918252519081900360200190f35b6000828211602a57508082036081566081565b6000565b5b92915050565b6000828284011115602a57508181016081566081565b6000565b5b9291505056"
+        "bytecode": "0x73142ce03e049269a37eb91743810194e1fcfeb16630146080604052600080fd00a165627a7a7230582083e86e46995b268179a35d3d89e28a4913a8104c518b63bc2b1aee04f9c0b6610029"
       },
-      "abi": [
-        {
-          "constant": false,
-          "inputs": [
-            {"name": "a","type": "uint256"},
-            {"name": "b","type": "uint256"}
-          ],
-          "name": "safeSub",
-          "outputs": [
-            {"name": "","type": "uint256"}
-          ],
-          "payable": false,
-          "type": "function"
-        },
-        {
-          "constant": false,
-          "inputs": [
-            {"name": "a","type": "uint256"},
-            {"name": "b","type": "uint256"}
-          ],
-          "name": "safeAdd",
-          "outputs": [
-            {"name": "","type": "uint256"}
-          ],
-          "payable": false,
-          "type": "function"
-        }
-      ],
+      "abi": [],
```

Comment:
ABI disappeared b/c the functions are now `internal` - therefore not part of the abi-interface -  which from what I can tell is library best practice.

Reply:
Hrm, that seems like something which is solidity specific.  It's probably inlining the actual function body into the contract that uses it which imho isn't an ideal extensible API since it would require that the compiler be able to compile the contract source.  This would disallow use of a Vyper library in a solidity contract or visa-versa.  I think these should remain externally callable.